lib/material_radio/material_radio_group.dart

```diff
@@ -60,106 +60,76 @@ import 'package:angular_components/utils/disposer/disposer.dart';
   final NgZone _ngZone;
-  final NgControl cd;
-  List<MaterialRadioComponent> _children;
-  bool _contentInit = false;
+  final _disposer = new Disposer.oneShot();
 
-  MaterialRadioGroupComponent(this._ngZone, @Self() @Optional() this.cd) {
-    _disposer.addStreamSubscription(componentSelection.selectionChanges
-        .listen((List<SelectionChangeRecord<MaterialRadioComponent>> changes) {
+  List<MaterialRadioComponent> _radioComponents = <MaterialRadioComponent>[];
+
+  MaterialRadioGroupComponent(this._ngZone, @Self() @Optional() NgControl cd) {
+    // When NgControl is present on the host element, the component participates
+    // in the Forms API.
+    cd?.valueAccessor = this;
+
+    _disposer.addStreamSubscription(
+        componentSelection.selectionChanges.listen((checkedChanges) {
       // Need to uncheck if selection change was made via user action.
-      for (var change in changes) {
-        for (MaterialRadioComponent item in change.removed) {
-          item.checked = false;
+      for (var checkedChange in checkedChanges) {
+        for (var radioComponent in checkedChange.removed) {
+          radioComponent.checked = false;
         }
       }
       // In case this was programmatically selected, thus won't get updated
       // by focus calls.
       _resetTabIndex();
-      MaterialRadioComponent selectedComponent =
-          componentSelection.selectedValues.isEmpty
-              ? null
-              : componentSelection.selectedValues.first;
-      _selected = selectedComponent == null ? null : selectedComponent.value;
+      _selected = _selectedRadioComponent?.value;
       if (valueSelection != null && _selected != null) {
         valueSelection.select(_selected);
       }
       _onChange.add(_selected);
     }));
 
-    _disposer.addStreamSubscription(
-        focusSelection.selectionChanges.listen((List changes) {
+    _disposer.addStreamSubscription(focusSelection.selectionChanges.listen((_) {
       _resetTabIndex();
     }));
-
-    // When NgControl is present on the host element, the component
-    // participates in the Forms API.
-    cd?.valueAccessor = this;
   }
 
-  @override
-  void ngAfterContentInit() {
-    _contentInit = true;
-    if (_preselectedValue != null) {
-      // Since this is updating children that were already dirty-checked,
-      // need to delay this change until next angular cycle.
-      _ngZone.onEventDone.first.then((_) {
-        if (_preselectedValue == null) return; // Overridden before callback.
-        // Initialize preselect now, this will trigger tabIndex reset.
-        selected = _preselectedValue;
-        // The preselected value should be used only once.
-        _preselectedValue = null;
-      });
-    } else {
-      // Initialize tabIndex.
-      _resetTabIndex();
-    }
-  }
+  @HostBinding('attr.role')
+  static const role = 'radiogroup';
+
+  @HostBinding('attr.tabindex')
+  static const tabIndex = -1;
```

Comment:
Should this be a string '-1'? (If so, I think that there are other instances where it is an int.)